python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -74,19 +77,37 @@ def get_nodes_to_launch(
             nodes: List of existing nodes in the cluster.
             pending_nodes: Summary of node types currently being launched.
             resource_demands: Vector of resource demands from the scheduler.
-            usage_by_ip: Mapping from ip to available resources.
+            unused_resources_by_ip: Mapping from ip to available resources.
             pending_placement_groups: Placement group demands.
-            static_node_resources: Mapping from ip to static node resources.
+            max_resources_by_ip: Mapping from ip to static node resources.
+            ensure_min_cluster_size: Try to ensure the cluster can fit at least
+                this set of resources. This differs from resources_demands in
+                that we don't take into account existing usage.
         """
+
+        # Calculate any extra requests we need to make that don't already fit
+        # into the cluster.
+        if ensure_min_cluster_size is not None:
+            used_resources = []
+            for ip, max_res in max_resources_by_ip.items():
```

Comment:
Something doesn't make sense here. Why we are not taking into account the pending launches or the nodes that are pending? If we have to take them into account then why not call `self.calculate_node_resources`?

Reply:
Because those are taken into account in the later calculation, this is just computing the resource delta needed to satisfy the request_resources() API.